examples/statemachine/src/language-server/statemachine-validator.ts

```diff
@@ -27,5 +32,35 @@ export class StatemachineValidator {
             }
         }
     }
-
+    /**
+     * Checks if there are duplicate state and event names.
+     * @param statemachine the statemachine to check
+     * @param accept the acceptor to report errors
+     */
+    checkUniqueStatesAndEvents(statemachine: Statemachine, accept: ValidationAcceptor): void {
+        // check for duplicate state and event names and add them to the map
+        const names = new Map<string, State | Event | undefined>();
```

Comment:
Minor: The return type of the `get` function of `Map` is already `T | undefined`. We don't need it explicitly here.
```suggestion
        const names = new Map<string, State | Event>();
```